HUD/hud_scraper.py

```diff
@@ -0,0 +1,177 @@
+        update_statement.group(1), "%B %Y"
+    ).date()
+    return scraped_date
+
+
+def check_site_for_new_date(existing_date):
+    """Check HUD web page with data files to see if the most
+    recently updated date is different than the date in MongoDB
+
+    Args:
+        existing_date (datetime): the date that the IRS last updated,
+        according to the data-sources collection
+
+    Returns:
+        bool: whether or not the dates are different
+    """
+    scraped_date = scrape_updated_date()
+    return scraped_date > existing_date
+
+
+def prevent_HUD_duplicates(pID, client, collection):
+    coll = client[collection]
+    dupe = coll.find_one(
+        {'Project ID': pID}
+    )
+    return dupe
+
+
+def purge_HUD_duplicates(df, client, collection, dupe_collection):
+    found_duplicates = []
+    for i in range(len(df)):
+        pID = int(df.loc[i, 'Project ID'])
+        if prevent_HUD_duplicates(pID, client, collection):
+            found_duplicates.append(i)
+    duplicate_df = df.loc[found_duplicates].reset_index(drop=True)
+    print('inserting tmpHUD dupes into the dupe collection')
+    insert_services(duplicate_df.to_dict('records'), client, dupe_collection)
+    df = df.drop(found_duplicates).reset_index(drop=True)
+    return df
+
+
+def main(client, check_collection, dump_collection, dupe_collection):
+    scraped_update_date = scrape_updated_date()
+    print(str(scraped_update_date))
+    try:
+        stored_update_date = client['data-sources'].find_one(
+            {"name": "hud_pit_hic_data"}
+        )['last_updated']
+        stored_update_date = datetime.strptime(
+            str(stored_update_date), '%Y-%m-%d %H:%M:%S'
+        ).date()
+        print(stored_update_date)
+        print(check_site_for_new_date(stored_update_date))
+        if not check_site_for_new_date(stored_update_date):
+            print('No new update detected. Exiting script...')
+            return
+    except KeyError:
+        print('Key Error')
+        pass
+    print('updating scraped update date in data-sources collection')
+    try:
+        client['data_sources'].update_one(
+            {"name": "irs_exempt_organizations"},
+            {'$set': {'last_updated': datetime.strftime(scraped_update_date, '%m/%d/%Y')}}
+        )
+    except errors.OperationFailure as e:
+        print(e)
+    df = grab_data()
+    print('purging duplicates from existing HUD collection')
+    if client[dump_collection].estimated_document_count() > 0:
+        df = df  # purge_HUD_duplicates(df, client, dump_collection, dupe_collection)
+    if client[check_collection].estimated_document_count() == 0:
+        # No need to check for duplicates in an empty collection
+        insert_services(df.to_dict('records'), client, dump_collection)
+    else:
+        print('refreshing ngrams')
+        # refresh_ngrams(client, check_collection)
+        found_duplicates = []
+        print('checking for duplicates in the services collection')
```

Comment:
should we also check duplicates with tmpIRS table because entry may not exist in services collection yet but might be present in tmpIRS which via volunteer portal will eventually make it to services collection. The same entry will also make it to services collection via tmpHUD also. I think we need to discuss this more, if entry is in both sources it gives me more confidence about it and we should probably track that information.